### PR TITLE
Add internal top tags API

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Feature

## Description

This PR adds an *internal* top tags API for suggestion purposes at `/tags/suggest`. I decided to not go with the one available at `/api/tags` for the following reasons:

1. The public API is still unstable which is why we versioned it as v0.
2. The public API returns attributes we won't need, like colors.
3. The public API sorts tags only by `taggings_count`. On the other hand, the Top Tags page (`/tags`) sorts them by "hotness score" which uses recently tagged articles and their comments, etc. During our call, we talked about "easy", "medium", and "hard" approaches to this topic, and reusing the logic from the internal controller gives us "medium" for free.
4. We may want to further tweak the logic in the future in ways that don't necessarily make sense for the public API.

Note: so far this endpoint only returns the tag names as an ordered array (index 0 is the "hottest" tag), if you'd like to have any other attributes available please let me know and I'll add them. For now, 100 tags will be returned, let me know if we should increase this number.

## Related Tickets & Documents

#14751 

## QA Instructions, Screenshots, Recordings

Nothing specific, the test should cover this quite well. Here's a request and the response:

API request example

### UI accessibility concerns?

None

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: without a frontend, this PR is quite meaningless by itself. We should write a changelog post once the whole user story is finished.

### CODE REVIEW

File: spec/requests/tags_spec.rb [hunk 1]
```diff
@@ -3,9 +3,20 @@
 RSpec.describe "Tags", type: :request, proper_status: true do
   describe "GET /tags" do
     it "returns proper page" do
-      get "/tags"
+      get tags_path
```
Comment (citizen428) [inline spec/requests/tags_spec.rb:6]: Since I was here already I updates this too.

**Comment by aitchiss** [comment]
> Note: so far this endpoint only returns the tag names as an ordered array (index 0 is the "hottest" tag), if you'd like to have any other attributes available please let me know and I'll add them. For now, 100 tags will be returned, let me know if we should increase this number.

In the current tag search, we show the `rules_html` in the suggestion too, so we should include that in this response 🙂

**Comment by citizen428** [comment]
@aitchiss I added `rules_html` to the API now.

File: app/controllers/tags_controller.rb [hunk 2]
```diff
@@ -4,11 +4,23 @@ class TagsController < ApplicationController
...
   ATTRIBUTES_FOR_SERIALIZATION = %i[id name bg_color_hex text_color_hex].freeze
+  INDEX_API_ATTRIBUTES = %i[name rules_html].freeze
 
   def index
     skip_authorization
     @tags_index = true
-    @tags = Tag.includes(:sponsorship).order(hotness_score: :desc).limit(100)
+    tags = Tag.order(hotness_score: :desc).limit(100)
+
+    respond_to do |format|
```
Comment (Ridhwana) [inline app/controllers/tags_controller.rb:14]: @citizen428 I have a general-ish question regarding this, it feels unusual that we return different info to the html format format vs the json format. I would have expected that the data that we return would be the same and only the format to differ, but maybe this is just the nature of the apps I've worked on. 

Is it a common practice to return different json vs html?

Comment (citizen428) [reply app/controllers/tags_controller.rb:14]: > I have a general-ish question regarding this, it feels unusual that we return different info to the html format format vs the json format.

That's a fair observation. In a way, we do return the same data, as we start from the same AR relation. Then for each format, we provide all the information that is necessary to display the information, which in the case of HTML is a bit more for the serverside rendering, whereas in the case of JSON it's just two attributes. Sending all attributes over the wire if we don't need them seemed a bit wasteful.

**Review by Ridhwana — APPROVED** [review]

File: app/controllers/tags_controller.rb (hunk 2 above)
Comment (citizen428) [reply app/controllers/tags_controller.rb]: We talked through this on Slack and in the end I decided to move it to a custom route.

**Review by Ridhwana — APPROVED** [review]
Thanks for the updates @citizen428!

**Review by aitchiss — APPROVED** [review]
LGTM 🚀 ✨

**Pull request merged**